apps/omg_watcher_info/lib/omg_watcher_info/db/eth_event.ex

```diff
@@ -209,16 +210,18 @@ defmodule OMG.WatcherInfo.DB.EthEvent do
             output_pointer: {:utxo_position, Utxo.Position.t()} | {:output_id, tuple()},
             eth_height: pos_integer()
           },
-          available_event_type_t()
-        ) :: {:ok, %__MODULE__{}} | {:error, Ecto.Changeset.t()} | :noop
+          available_event_type_t(),
+          boolean()
+        ) :: :ok | :noop
   defp insert_exit!(
          %{
            root_chain_txhash: root_chain_txhash,
            log_index: log_index,
            eth_height: eth_height,
            output_pointer: output_pointer
          },
-         event_type
+         event_type,
```

Comment:
maybe at this point it also make sense to refactor the function header by not pattern match in it

Reply:
I think extracting the event values is justified as we need all of them few lines below in `%__MODULE__{...}`

Reply:
yeah, but we're trying to get away from the code style where pattern matching is being done in function headers

```
defp insert_exit!(event, event_type, ensure_output) do
%{root_chain_txhash: root_chain_txhash, log_index: log_index, eth_height: eth_height, output_pointer: output_pointer} = event
root_chain_txhash_event = generate_root_chain_txhash_event(root_chain_txhash, log_index)
...
```

the module becomes much more readable this way.

Reply:
Considering formatter "help" I have a mixed feeling about readability... but at least it's closer to the convention